### DISCUSSION TITLE
Search results list menu

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#540] [Q&A] [answered]
When searching for a Favorite/POI and we want to navigate to it, there's often no need to tap it to display the icon+info on the map if we are happy that we know we've got the right one.

How about being able to long-press on the search result list to invoke the same popup menu that you get when you long-press a Favorite icon on the map?

**Comment by devemux86** [comment]
@MotoUKRider thanks for the suggestion.

When we select a search result it appears on the map. Then we can long press it and use it through its menu.
Do you mean we could use a different method?

We could also add a menu icon to the right of each result, like in other lists.

**Reply by MotoUKRider** [reply]
@devemux86 Yes, I find the "icon on the map" annoying, I was hoping that we could just select it from the drop-down list.  The 3 dots on the other lists are a *little* small, but manageable!

**Comment by devemux86** [comment]
Next Cruiser will display a menu button in the search results that will immediately open their action menu.
(like when we select them on the map)

<img src="https://github.com/devemux86/cruiser/assets/3484020/76d678e0-66e7-49e0-b1c8-3ba4062bc5ba" width="50%"/>

**Comment by devemux86** [comment]
Cruiser now displays a menu button in the search results that opens their action menu:

- https://github.com/devemux86/cruiser/discussions/610

**Comment by DolomitesTom** [comment]
yes this works now, but:

The search routine somehow doesn't work, example "Gerolstein", on a first search it finds nothing, on a second search it finds some streets named "Gerolsteiner Straße" in Frankfurt. On further searches it finds it. Something is not ok (search areas active only online, routes, favorites)

**Reply by devemux86** [reply]
@DolomitesTom you mentioned using online search, OpenRouteService or GraphHopper?

Both services work here, _if the searched item is inside the visible map area_.

It works like you asked here:

- https://github.com/devemux86/cruiser/discussions/604#discussioncomment-10141004

> if it is possible to have online search also inside the visible map area

**Comment by DolomitesTom** [comment]
OpenRouteService, but it is the same with Graphhopper. In Cruiser 5 "Gerolstein" finds nothing. In Cruiser 4 it finds the city online. Did you test this?

**Reply by devemux86** [reply]
As mentioned above, both online search services work correctly,
if the searched item is inside the visible map area (as requested).

**Reply by DolomitesTom** [reply]
ok, this is good for not listed categories, like Biergarten. But what if I want to find a city and street, full address or other location which is not in the actual map area in order to include it in the route planning? Shouldn't this work in all cases, at least in the online search?

**Reply by devemux86** [reply]
Here you asked:

> if it is possible to have online search also inside the visible map area

- https://github.com/devemux86/cruiser/discussions/604#discussioncomment-10141004

If you want the previous workflow we can restore it, but you should decide what you want? :slightly_smiling_face:

**Reply by devemux86** [reply]
> at least in the online search?

In online search it is possible, like before, because the search is done in online powerful desktop servers.

In offline search (done locally on the phone), we do not enable it for performance reasons.

**Reply by DolomitesTom** [reply]
I think you misunderstand. Online search for categories within the visible map is fine. But the online search for addresses should work also outside of the visible map. Maybe I was not precise enough when formulating the request, my fault. I don't know how you could identify it, maybe by splitting the search in category-search and full search?

**Reply by devemux86** [reply]
That's why reports need to be very very detailed, so that we can understand what they mean
and not waste time going backwards and forwards. :slightly_smiling_face: 

---

- Online free text search will be restored without bounds, like before.

- Online category search always uses the visible map area.

**Reply by DolomitesTom** [reply]
ok but "Biergarten" is no existing category, but only a virtual category. How would you recognize that I only want to see them in the visible map?

**Reply by devemux86** [reply]
@DolomitesTom if you mean [amenity=biergarten](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbiergarten), this is included in the "Food" category.

**Reply by devemux86** [reply]
Or search with text "biergarten" with offline or online POIs,
but have to zoom in enough and search in the visible area.

Cannot search for POIs in the whole Germany.

**Comment by devemux86** [comment, marked as answer]
- A menu button appears in the search results that opens their action menu in https://github.com/devemux86/cruiser/discussions/641
